projects/imagemagick/crop_fuzzer.cc

```diff
@@ -0,0 +1,21 @@
+#include <cstdint>
+
+#include <Magick++/Blob.h>
+#include <Magick++/Image.h>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+  if (Size < 4) {
+    return 0;
+  }
+  size_t Width = *reinterpret_cast<const uint16_t *>(Data);
+  size_t Height = *reinterpret_cast<const uint16_t *>(Data + 2);
+  const Magick::Blob blob(Data + 4, Size - 4);
```

Comment:
I'd like to suggest another way of generating width and height: calculate a `std::hash` or some CRC value out of `(Data, Size)`, then take lower and higher 16 bits of the hash as width and height value. That way, we can simply put valid streams into the corpus, plus don't have to mess with magic numbers such as 2, 4, etc, as `sizeof(uint16_t)` is not guaranteed to be 2 bytes AFAIK.

Reply:
The downside of using a hash is that the fuzzing engine is no longer intelligently able to modify the geometry independently of modifying the image. This seems like it makes it harder for the AI^W fuzzing engine.

Reply:
That's a fair point, let's see what @kcc thinks. What kind of data do we need to pass to `Magick::Blob` constructor?

Reply:
Blob is constructed from arbitrary bytes.